packages/components/src/components/trace/TraceFilter.vue

```diff
@@ -0,0 +1,391 @@
+
+        if (selected.offsetTop < this.$refs.suggestionsList.scrollTop) {
+          this.$refs.suggestionsList.scrollTop = selected.offsetTop;
+        } else if (
+          selected.offsetTop >=
+          this.$refs.suggestionsList.scrollTop +
+            this.$refs.suggestionsList.offsetHeight -
+            2
+        ) {
+          this.$refs.suggestionsList.scrollTop =
+            selected.offsetTop - selected.offsetHeight * 4;
+        }
+      },
+    },
+  },
+
+  computed: {
+    formClasses() {
+      const classList = ['trace-filter__form'];
+
+      if (this.showSuggestions && this.suggestions.length) {
+        classList.push('trace-filter__form--show-suggestions');
+      }
+
+      return classList;
+    },
+    suggestionsList() {
+      let suggestions = this.suggestions
+        .map((e) => e.toString())
+        .filter((e, i, arr) => arr.indexOf(e) === i);
+
+      const terms = this.filterValue.match(/(?:[^\s"]+|"[^"]*")+/g);
+      const lastTerm = terms ? terms[terms.length - 1] : null;
+      if (
+        lastTerm &&
+        !/^".+"$/g.test(lastTerm) &&
+        !/^%.+%$/g.test(lastTerm) &&
+        !/^id:.+$/g.test(lastTerm) &&
+        !/^label:.+$/g.test(lastTerm)
+      ) {
+        suggestions = suggestions.filter((item) =>
+          new RegExp(lastTerm, 'ig').test(item)
+        );
+      }
+
+      return suggestions;
+    },
+  },
+
+  methods: {
+    setValue(newValue) {
+      this.filterValue = newValue;
+    },
+    clearValue() {
+      this.filterValue = '';
+    },
+    onPrevArrow() {
+      this.$emit('onPrevArrow');
+    },
+    onNextArrow() {
+      this.$emit('onNextArrow');
+    },
+    getSuggestionClasses(index) {
+      const classes = ['trace-filter__suggestions-item'];
+      if (index === this.selectedSuggestion) {
+        classes.push('trace-filter__suggestions-item--selected');
+      }
+      return classes;
+    },
+    onInputFocus() {
+      this.showSuggestions = true;
+      this.selectedSuggestion = 0;
+    },
+    onFormSubmit() {
+      if (
+        this.$refs.input === window.document.activeElement &&
+        this.showSuggestions &&
+        this.suggestionsList.length
+      ) {
+        this.makeSelection(
+          this.suggestionsList[this.selectedSuggestion].toString()
+        );
+      }
+    },
+    makeSelection(eventName) {
+      const terms = this.filterValue.match(/(?:[^\s"]+|"[^"]*")+/g);
+      const lastTerm = terms ? terms[terms.length - 1] : null;
+      if (
+        lastTerm &&
+        !/^".+"$/g.test(lastTerm) &&
+        !/^%.+%$/g.test(lastTerm) &&
+        !/^id:.+$/g.test(lastTerm) &&
+        !/^label:.+$/g.test(lastTerm)
+      ) {
+        terms.pop();
+      }
+      this.filterValue = `${
+        terms ? terms.join(' ') : ''
+      } "${eventName}"`.trim();
+    },
+    onWindowClick(event) {
+      if (!getEventTarget(event.target, this.$refs.form)) {
+        this.showSuggestions = false;
```

Comment:
I think this would best be done upon losing keyboard focus. Otherwise, a drag selection is likely to close the suggestion dropdown.

https://user-images.githubusercontent.com/8737782/153258684-1dc0c73c-b810-4e22-beb9-91df773fa078.mp4

Reply:
Seems there is a bug in Chrome with drag selection of input contents.
(In Firefox `click` event is not triggered when I finish selection outside input)
Fixed https://github.com/applandinc/appmap-js/pull/534/commits/1ee196b1ebf066702e08604d77911effb3f55dda